sshuttle/methods/pf.py

```diff
@@ -108,16 +111,21 @@ def pf_query_nat(family, proto, src_ip, src_port, dst_ip, dst_port):
     pnl.direction = PF_OUT
     pnl.af = family
     memmove(addressof(pnl.saddr), packed_src_ip, length)
-    pnl.sxport.port = socket.htons(src_port)
     memmove(addressof(pnl.daddr), packed_dst_ip, length)
-    pnl.dxport.port = socket.htons(dst_port)
+    if sys.platform == 'darwin':
+        pnl.sxport.port = socket.htons(src_port)
+        pnl.dxport.port = socket.htons(dst_port)
+    else:
+        pnl.sxport = socket.htons(src_port)
+        pnl.dxport = socket.htons(dst_port)
```

Comment:
Oh, I see, you have redefined `pf_state_xport` to `c_uint16`. Maybe instead of completely replacing it, you could just drop the spi field, hence it should become 16bits (I assume this is why you redefined it).